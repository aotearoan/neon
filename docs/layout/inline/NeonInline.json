{
  "displayName": "NeonInline",
  "description": "A horizontal layout component. NeonInline provides a way of laying out it's contents with standard gaps. At the\nspecified breakpoint the layout will wrap to vertical.",
  "tags": {},
  "exportName": "default",
  "props": [
    {
      "name": "gap",
      "description": "Size of the gap between items",
      "type": {
        "name": "String as () => NeonSize"
      },
      "defaultValue": {
        "func": true,
        "value": "() => NeonSize.Large"
      }
    },
    {
      "name": "breakpoint",
      "description": "Breakpoint at which to layout switches to vertical. TIP: Pass <em>breakpoint=\"\"</em> to override the default\nbreakpoint & prevent switching to a vertical layout.",
      "type": {
        "name": "String as () => NeonSize"
      },
      "defaultValue": {
        "func": true,
        "value": "() => NeonResponsive.MobileLarge"
      }
    }
  ],
  "slots": [
    {
      "name": "default",
      "description": "stack contents"
    }
  ]
}