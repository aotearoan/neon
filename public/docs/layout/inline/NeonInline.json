{
  "displayName": "NeonInline",
  "description": "A horizontal layout component. NeonInline provides a way of laying out it's contents with standard gaps. At the\nspecified breakpoint the layout will wrap to vertical.",
  "tags": {},
  "exportName": "default",
  "props": [
    {
      "name": "gap",
      "description": "Size of the gap between items",
      "type": {
        "name": "String as () => NeonSize"
      },
      "defaultValue": {
        "func": true,
        "value": "() => NeonSize.Large"
      }
    },
    {
      "name": "breakpoint",
      "description": "Breakpoint at which to layout wraps to vertical",
      "type": {
        "name": "String as () => NeonSize"
      },
      "defaultValue": {
        "func": true,
        "value": "() => NeonResponsive.MobileLarge"
      }
    }
  ],
  "slots": [
    {
      "name": "default",
      "description": "stack contents"
    }
  ]
}