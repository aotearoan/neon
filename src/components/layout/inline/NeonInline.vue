<template>
  <div :class="[`neon-inline--gap-${gap}`, `neon-inline--breakpoint-${breakpoint}`]" class="neon-inline">
    <!-- @slot stack contents -->
    <slot></slot>
  </div>
</template>

<script lang="ts" src="./NeonInline.ts"></script>
