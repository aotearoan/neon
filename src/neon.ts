/* components */
export { default as NeonActionMenu } from './components/navigation/action-menu/NeonActionMenu.vue';
export { default as NeonAlert } from './components/feedback/alert/NeonAlert.vue';
export { default as NeonAlertContainer } from './components/feedback/alert/container/NeonAlertContainer.vue';
export { default as NeonAnchor } from './components/navigation/anchor/NeonAnchor.vue';
export { default as NeonBadge } from './components/presentation/badge/NeonBadge.vue';
export { default as NeonButton } from './components/user-input/button/NeonButton.vue';
export { default as NeonCard } from './components/layout/card/NeonCard.vue';
export { default as NeonCardBody } from './components/layout/card/body/NeonCardBody.vue';
export { default as NeonCardFooter } from './components/layout/card/footer/NeonCardFooter.vue';
export { default as NeonCardHeader } from './components/layout/card/header/NeonCardHeader.vue';
export { default as NeonChip } from './components/user-input/chip/NeonChip.vue';
export { default as NeonColor } from './components/user-input/color/NeonColor.vue';
export { default as NeonDialog } from './components/feedback/dialog/NeonDialog.vue';
export { default as NeonDrawer } from './components/layout/drawer/NeonDrawer.vue';
export { default as NeonDropdown } from './components/presentation/dropdown/NeonDropdown.vue';
export { default as NeonDropdownMenu } from './components/navigation/dropdown-menu/NeonDropdownMenu.vue';
export { default as NeonDropZone } from './components/user-input/drop-zone/NeonDropZone.vue';
export {
  default as NeonExpansionIndicator
} from './components/presentation/expansion-indicator/NeonExpansionIndicator.vue';
export { default as NeonExpansionPanel } from './components/presentation/expansion-panel/NeonExpansionPanel.vue';
export { default as NeonFieldGroup } from './components/user-input/field-group/NeonFieldGroup.vue';
export { default as NeonFile } from './components/user-input/file/NeonFile.vue';
export { default as NeonFilterList } from './components/user-input/filter-list/NeonFilterList.vue';
export { default as NeonFooter } from './components/layout/footer/NeonFooter.vue';
export { default as NeonGrid } from './components/layout/grid/NeonGrid.vue';
export { default as NeonGridArea } from './components/layout/grid/grid-area/NeonGridArea.vue';
export { default as NeonIcon } from './components/presentation/icon/NeonIcon.vue';
export { default as NeonInput } from './components/user-input/input/NeonInput.vue';
export { default as NeonInputIndicator } from './components/user-input/input-indicator/NeonInputIndicator.vue';
export { default as NeonLabel } from './components/presentation/label/NeonLabel.vue';
export { default as NeonLinearProgress } from './components/feedback/linear-progress/NeonLinearProgress.vue';
export { default as NeonLink } from './components/navigation/link/NeonLink.vue';
export { default as NeonList } from './components/user-input/list/NeonList.vue';
export { default as NeonLogo } from './components/presentation/logo/NeonLogo.vue';
export { default as NeonMenu } from './components/navigation/menu/NeonMenu.vue';
export { default as NeonModal } from './components/layout/modal/NeonModal.vue';
export { default as NeonNote } from './components/feedback/note/NeonNote.vue';
export {
  default as NeonNotificationCounter
} from './components/feedback/notification-counter/NeonNotificationCounter.vue';
export { default as NeonNumber } from './components/user-input/number/NeonNumber.vue';
export { default as NeonPage } from './components/layout/page/NeonPage.vue';
export { default as NeonPassword } from './components/user-input/password/NeonPassword.vue';
export { default as NeonRangeSlider } from './components/user-input/range-slider/NeonRangeSlider.vue';
export { default as NeonSideNav } from './components/layout/side-nav/NeonSideNav.vue';
export { default as NeonSearch } from './components/user-input/search/NeonSearch.vue';
export { default as NeonSelect } from './components/user-input/select/NeonSelect.vue';
export { default as NeonSkeletonLoader } from './components/feedback/skeleton-loader/NeonSkeletonLoader.vue';
export { default as NeonSlider } from './components/user-input/slider/NeonSlider.vue';
export { default as NeonSplashLoader } from './components/feedback/splash-loader/NeonSplashLoader.vue';
export { default as NeonSwitch } from './components/user-input/switch/NeonSwitch.vue';
export { default as NeonTab } from './components/presentation/tabs/tab/NeonTab.vue';
export { default as NeonTabs } from './components/presentation/tabs/NeonTabs.vue';
export { default as NeonToastContainer } from './components/feedback/alert/container/NeonToastContainer.vue';
export { default as NeonToggle } from './components/user-input/toggle/NeonToggle.vue';
export { default as NeonToggleChip } from './components/user-input/toggle-chip/NeonToggleChip.vue';
export { default as NeonTooltip } from './components/feedback/tooltip/NeonTooltip.vue';
export { default as NeonTopNav } from './components/layout/top-nav/NeonTopNav.vue';
export { default as NeonTreeMenu } from './components/navigation/tree-menu/NeonTreeMenu.vue';

/* enums */
export { NeonAlertLevel } from './common/enums/NeonAlertLevel';
export { NeonAlertPlacement } from './common/enums/NeonAlertPlacement';
export { NeonButtonSize } from './common/enums/NeonButtonSize';
export { NeonButtonStyle } from './common/enums/NeonButtonStyle';
export { NeonChipAction } from './common/enums/NeonChipAction';
export { NeonDropdownPlacement } from './common/enums/NeonDropdownPlacement';
export { NeonDropdownStyle } from './common/enums/NeonDropdownStyle';
export { NeonFunctionalColor } from './common/enums/NeonFunctionalColor';
export { NeonHorizontalPosition } from './common/enums/NeonHorizontalPosition';
export { NeonInputMode } from './common/enums/NeonInputMode';
export { NeonInputType } from './common/enums/NeonInputType';
export { NeonLabelSize } from './common/enums/NeonLabelSize';
export { NeonMode } from './common/enums/NeonMode';
export { NeonOrientation } from './common/enums/NeonOrientation';
export { NeonOutlineStyle } from './common/enums/NeonOutlineStyle';
export { NeonPlacement } from './common/enums/NeonPlacement';
export { NeonPosition } from './common/enums/NeonPosition';
export { NeonResponsive } from './common/enums/NeonResponsive';
export { NeonSize } from './common/enums/NeonSize';
export { NeonSplashLoaderSize } from './common/enums/NeonSplashLoaderSize';
export { NeonState } from './common/enums/NeonState';
export { NeonSwitchStyle } from './common/enums/NeonSwitchStyle';
export { NeonToggleStyle } from './common/enums/NeonToggleStyle';
export { NeonTooltipStyle } from './common/enums/NeonTooltipStyle';
export { NeonVerticalPosition } from './common/enums/NeonVerticalPosition';

/* models */
export type { NeonActionMenuModel } from './common/models/NeonActionMenuModel';
export type { NeonAlertAction } from './common/models/NeonAlertAction';
export type { NeonAlertMessage } from './common/models/NeonAlertMessage';
export type { NeonAlertModel } from './components/feedback/alert/NeonAlertModel';
export type { NeonAvailableSpace } from './common/models/NeonAvailableSpace';
export type { NeonDropdownMenuItem } from './common/models/NeonDropdownMenuItem';
export type { NeonDropdownPlacementObject } from './common/models/NeonDropdownPlacementObject';
export type { NeonFilterListItem } from './common/models/NeonFilterListItem';
export type { NeonFormattedDate } from './common/models/NeonFormattedDate';
export type { NeonGridModel } from './common/models/NeonGridModel';
export type { NeonListItem } from './common/models/NeonListItem';
export type { NeonMenuItem } from './common/models/NeonMenuItem';
export type { NeonMenuModel } from './common/models/NeonMenuModel';
export type { NeonNumberFormatOptions } from './common/models/NeonNumberFormatOptions';
export type { NeonPriorityMenuItem } from './components/navigation/menu/NeonPriorityMenuItem';
export type { NeonSearchOption } from './common/models/NeonSearchOption';
export type { NeonSelectGroup } from './common/models/NeonSelectGroup';
export type { NeonSelectOption } from './common/models/NeonSelectOption';
export type { NeonTabModel } from './common/models/NeonTabModel';
export type { NeonToastMessage } from './common/models/NeonToastMessage';
export type { NeonToastModel } from './components/feedback/alert/NeonToastModel';
export type { NeonToggleModel } from './common/models/NeonToggleModel';
export type { NeonTreeMenuLinkModel, NeonTreeMenuSectionModel } from './common/models/NeonTreeMenuModel';

/* utils */
export { NeonAlertService } from './common/utils/NeonAlertService';
export { NeonClipboardService, NeonClipboardSupport } from './common/utils/NeonClipboardService';
export { NeonClosableUtils } from './common/utils/NeonClosableUtils';
export { NeonDateUtils } from './common/utils/NeonDateUtils';
export { NeonDropdownPlacementUtils } from './common/utils/NeonDropdownPlacementUtils';
export { NeonEventBus } from './common/utils/NeonEventBus';
export { NeonIconRegistry } from './common/utils/NeonIconRegistry';
export { NeonModeUtils } from './common/utils/NeonModeUtils';
export { NeonNumberUtils } from './common/utils/NeonNumberUtils';
export { NeonPlacementUtils } from './common/utils/NeonPlacementUtils';
export { NeonResponsiveUtils } from './common/utils/NeonResponsiveUtils';
export { NeonScrollUtils } from './common/utils/NeonScrollUtils';
export { NeonToastService } from './common/utils/NeonToastService';
export { NeonTooltipPlacementUtils } from './common/utils/NeonTooltipPlacementUtils';

/* register default provided icons */
export { RegisterIcons } from './common/utils/RegisterIcons';
