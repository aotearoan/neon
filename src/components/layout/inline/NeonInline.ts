import { defineComponent } from 'vue';
import { NeonSize } from '@/common/enums/NeonSize';
import { NeonResponsive } from '@/neon';

/**
 * A horizontal layout component. NeonInline provides a way of laying out it's contents with standard gaps. At the
 * specified breakpoint the layout will wrap to vertical.
 */
export default defineComponent({
  name: 'NeonInline',
  props: {
    /**
     * Size of the gap between items
     */
    gap: { type: String as () => NeonSize, default: () => NeonSize.Large },
    /**
     * Breakpoint at which to layout wraps to vertical
     */
    breakpoint: { type: String as () => NeonSize, default: () => NeonResponsive.MobileLarge },
  },
});
