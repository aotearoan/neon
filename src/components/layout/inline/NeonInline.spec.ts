import type { RenderResult } from '@testing-library/vue';
import { render } from '@testing-library/vue';
import NeonInline from './NeonInline.vue';
import { NeonResponsive, NeonSize } from '@/neon';

describe('NeonInline', () => {
  let harness: RenderResult;

  beforeEach(() => {
    harness = render(NeonInline, {
      props: {},
      slots: { default: '<p>test</p>' },
    });
  });

  it('renders slot contents', () => {
    // given
    const { html } = harness;
    // when / then
    expect(html()).toMatchSnapshot();
  });

  it('renders default gap', () => {
    // given
    const { html } = harness;
    // when / then
    expect(html()).toMatch('neon-inline--gap-l');
  });

  it('renders gap', async () => {
    // given
    const { html, rerender } = harness;
    await rerender({ gap: NeonSize.Small });
    // when / then
    expect(html()).toMatch('neon-inline--gap-s');
  });

  it('renders default breakpoint', () => {
    // given
    const { html } = harness;
    // when / then
    expect(html()).toMatch('neon-inline--breakpoint-mobile-large');
  });

  it('renders breakpoint', async () => {
    // given
    const { html, rerender } = harness;
    await rerender({ breakpoint: NeonResponsive.Tablet });
    // when / then
    expect(html()).toMatch('neon-inline--breakpoint-tablet');
  });
});
