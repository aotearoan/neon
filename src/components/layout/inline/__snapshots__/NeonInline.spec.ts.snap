// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NeonInline renders slot contents 1`] = `
<div class="neon-inline--gap-l neon-inline--breakpoint-mobile-large neon-inline">
  <!-- @slot stack contents -->
  <p>test</p>
</div>
`;
